src/hooks/useInfiniteScroll.jsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef, useState } from "react";
+
+const useInfiniteScroll = (posts) => {
+	const [pageNumber, setPageNumber] = useState(1);
+	const lastElementReference = useRef(null);
+	const postsLength = posts.length;
+	const hasMorePosts = pageNumber <= Math.ceil(postsLength / pageNumber);
```

Comment:
This looks interesting. Does this have any special name? I've never used this arrow in reversed manner.